src/averageAges.js

```diff
@@ -55,7 +72,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothers = people.filter(person => person.sex === 'f'
+    && (people.some(human => human.mother === person.name)));
+
+  const children = people
+    .filter(child => mothers.some(mother => child.mother === mother.name)
+    && (onlyWithSon ? child.sex === 'm' : true));
+
+  const ageDifferences = children.map(child => {
+    const foundMom = mothers.find(mom => mom.name === child.mother);
+
+    return foundMom ? child.born - foundMom.born : null;
+  });
+
+  const averageDiff = ageDifferences
+    .reduce((sum, char) => sum + char) / ageDifferences.length;
+
+  return averageDiff;
```

Comment:
follow the same logic for return, when you calculate everything before return do it everywhere